src/include/common/serializer/buffered_file.h

```diff
@@ -41,6 +42,13 @@ class BufferedFileReader final : public Reader {
 public:
     explicit BufferedFileReader(std::unique_ptr<FileInfo> fileInfo);
 
+    // Note: this function is resting the next file offset to read. Make sure the buffer is empty.
```

Comment:
I think the "Make sure the buffer is empty." message can be omitted here since it's resetting the bufferOffset and bufferSize. The other function wasn't supposed to be called when there's something in the buffer, but this one wouldn't have the same issue.